File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/abstractfactory/trasnport/FourWheelTransportFactory.java
package com.bigboxcode.designpattern.abstractfactory.trasnport;

public class FourWheelTransportFactory implements AbstractTransportFactory{
    @Override
    public Transport getTransport(String type) {

        if (type.equalsIgnoreCase("car")) {
            return new Car();
        }

        if (type.equalsIgnoreCase("truck")) {
            return new Truck();
        }

        return null;
    }
}

File: design-patterns/php/abstractfactory/transport/FourWheelTransportFactory.php
<?php
// FourWheelTransportFactory.php

namespace BigBoxCode\DesignPattern\AbstractFactory\Transport;

class FourWheelTransportFactory implements AbstractTransportFactory {
    public function getTransport(string $type): ?Transport {
        if (strcasecmp($type, "car") === 0) {
            return new Car();
        }

        if (strcasecmp($type, "truck") === 0) {
            return new Truck();
        }

        return null;
    }
}